client/src/components/Loading.tsx

```diff
@@ -1,10 +1,22 @@
+import { Suspense } from "react";
+import { Await, useLoaderData, Outlet, Navigate } from "react-router-dom";
 import { Box, CircularProgress } from "@chakra-ui/react";
 
 const Loading = () => {
+  const data = useLoaderData() as { response: null };
```

Comment:
nitpick: our loader's response value is an object with a `Promise` in the `response` field, and that `Promise` returns null, so maybe we should reflect that in the type annotation?